Cython/Includes/libcpp/string_view.pxd

```diff
@@ -0,0 +1,181 @@
+from libcpp.string cimport string
+
+cdef extern from "<string_view>" namespace "std::string_view" nogil:
+    const size_t npos
+
+cdef extern from "<string_view>" namespace "std" nogil:
+
+    cdef cppclass string_view:
+        ctypedef char value_type
+        ctypedef size_t size_type
+        ctypedef ptrdiff_t difference_type
+
+        bint operator==(string_view)
+        bint operator!= (string_view)
+        bint operator< (string_view)
+        bint operator> (string_view)
+        bint operator<= (string_view)
+        bint operator>= (string_view)
+
+        cppclass const_iterator:
+            const_iterator()
+            const_iterator(const_iterator&)
+            operator=(const_iterator&)
+            const value_type& operator*()
+            const_iterator operator++()
+            const_iterator operator--()
+            const_iterator operator++(int)
+            const_iterator operator--(int)
+            const_iterator operator+(size_type)
+            const_iterator operator-(size_type)
+            difference_type operator-(const_iterator)
+            bint operator==(const_iterator)
+            bint operator!=(const_iterator)
+            bint operator<(const_iterator)
+            bint operator>(const_iterator)
+            bint operator<=(const_iterator)
+            bint operator>=(const_iterator)
+
+        ctypedef const_iterator iterator
+
+        cppclass const_reverse_iterator:
+            const_reverse_iterator()
+            const_reverse_iterator(const_reverse_iterator&)
+            operator=(const_reverse_iterator&)
+            const value_type& operator*()
+            const_reverse_iterator operator++()
+            const_reverse_iterator operator--()
+            const_reverse_iterator operator++(int)
+            const_reverse_iterator operator--(int)
+            const_reverse_iterator operator+(size_type)
+            const_reverse_iterator operator-(size_type)
+            difference_type operator-(const_reverse_iterator)
+            bint operator==(const_reverse_iterator)
+            bint operator!=(const_reverse_iterator)
+            bint operator<(const_reverse_iterator)
+            bint operator>(const_reverse_iterator)
+            bint operator<=(const_reverse_iterator)
+            bint operator>=(const_reverse_iterator)
+
+        ctypedef const_reverse_iterator reverse_iterator
+
+        # Constructors
+        string_view()  # (1)
+        string_view(string_view s)  # (2)
+        string_view(const char* s, size_t n)  # (3)
+        string_view(const char* s)  # (4)
+        string_view& string_view[InputIt](InputIt, InputIt)  # (5)
+
+        # This should be `operator string_view()` on std::string, but
+        # Cython doesn't yet allow declaring it.
+        string_view(const string&)
```

Comment:
Does this look right @da-woods ?

Reply:
We've got `bool operator bool()` declarations elsewhere. Which makes me thing it is possible to declare operators like that. Although maybe bool is special-cased.

With that said, I'm fairly sure we don't support the implicit conversion so this is as clear as anything.